SteamKit2/SteamKit2/Types/DepotManifest.cs

```diff
@@ -226,16 +226,16 @@ public bool DecryptFilenames(byte[] encryptionKey)
         /// <returns><c>true</c> if serialization was successful; otherwise, <c>false</c>.</returns>
         public bool SaveToFile( string filename )
         {
-            using var fs = File.Open( filename, FileMode.Create );
-            using var bw = new BinaryWriter( fs );
-            var data = Serialize();
-            if ( data != null )
+            try
             {
-                bw.Write( data );
-                return true;
+                using var fs = File.Open( filename, FileMode.Create );
+                Serialize( fs ); // Directly pass the FileStream to the Serialize method
+                return true; // If serialization completes without throwing an exception, return true
+            }
+            catch ( Exception )

```

Comment:
Are we expecting any exceptions, why the catch?

Reply:
I think these should just throw.

Reply:
`catch (Exception)` is also bad practice: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1031

Either we should catch certain known exceptions, or let them leak out in actually-exceptional scenarios.

Reply:
I also can't find a way in which the old method would ever return `false` because `Serialize()` seems to always return a non-null value.